compiler/rustc_middle/src/mir/tcx.rs

```diff
@@ -223,6 +223,23 @@ impl<'tcx> Rvalue<'tcx> {
             _ => RvalueInitializationState::Deep,
         }
     }
+
+    pub fn is_pointer_int_cast<D>(&self, local_decls: &D, tcx: TyCtxt<'tcx>) -> bool
+    where
+        D: HasLocalDecls<'tcx>,
+    {
+        if let Rvalue::Cast(CastKind::Misc, src_op, dest_ty) = self {
+            if *dest_ty == tcx.types.isize || *dest_ty == tcx.types.usize {
```

Comment:
Interesting. I've also discovered that there's `CastTy`, which sort of solves this same problem as well. Not sure what to use now. I guess I'll switch to `CastTy` for now, since that seems to be what most backends use?

Reply:
Hm, Miri doesn't use `CastTy`... maybe it should?^^